make/targets.mk

```diff
@@ -1,58 +1,13 @@
-OFFICIAL_TARGETS  = ALIENFLIGHTF3 ALIENFLIGHTF4 ANYFCF7 BETAFLIGHTF3 BLUEJAYF4 FURYF4 REVO SIRINFPV SPARKY SPRACINGF3 SPRACINGF3EVO SPRACINGF3NEO SPRACINGF4EVO SPRACINGF7DUAL STM32F3DISCOVERY
-ALT_TARGETS       = $(sort $(filter-out target, $(basename $(notdir $(wildcard $(ROOT)/src/main/target/*/*.mk)))))
-NOBUILD_TARGETS   = $(sort $(filter-out target, $(basename $(notdir $(wildcard $(ROOT)/src/main/target/*/*.nomk)))))
-OPBL_TARGETS      = $(filter %_OPBL, $(ALT_TARGETS))
-
-VALID_TARGETS   = $(dir $(wildcard $(ROOT)/src/main/target/*/target.mk))
-VALID_TARGETS  := $(subst /,, $(subst ./src/main/target/,, $(VALID_TARGETS)))
-VALID_TARGETS  := $(VALID_TARGETS) $(ALT_TARGETS)
-VALID_TARGETS  := $(sort $(VALID_TARGETS))
-VALID_TARGETS  := $(filter-out $(NOBUILD_TARGETS), $(VALID_TARGETS))
+include $(ROOT)/make/targets_list.mk
```

Comment:
Can you please explain why this was split?

Also, I think this include should be in the main `Makefile`, and not as an indirect include.

Reply:
The original `targets.mk` combined listing of targets with a consistency check that the chosen target is viable, i.e., it assumed the choice of a particular target.

For the new per-target test, target listing code is needed, but no single target (or base target) is picked. Instead, the list of all targets is expanded into per-target instances of a test.

So I moved the target listing code into a reusable `targets_list.mk` fragment shared by the test `Makefile` and by `targets.mk` — to keep listing wildcards in one place and accommodate both use cases mentioned above.

As for the indirect include, `targets.mk` depends on `targets_list.mk`. Including `targets_list.mk` directly in the main `Makefile` (and not in in `targets.mk`) would make `targets.mk` less “self-contained”; its users would need to include (or otherwise provide) `$(get_base_target ...)`, for example.

Reply:
Ok, makes sense.